### ISSUE TITLE
Removing an automatic added JVM brings it back after restart

### ISSUE THREAD

**Issue by laeubi** [eclipse-jdt/eclipse.jdt.core#1293]
I have a "bad" JVM that is discovered on the system but removing it has no permanent effect: After a restart, the removed JVM is added back.

FYI @mickaelistria I think it is important to store somewhere if a user has explicit removed a JVM (e.g. by its path) so they do not appear again and again. Currently the only way is disable the feature all together or uninstall/remove the discovered JVM.

**Comment by HannesWell** [comment]
I also had troubles with the auto-search of VMs because I have jextract in my java/jdk folder and therefore the JVM contained in jextract is also found.
But jextract only contains a reduced JVM (probably build with jlink) and once it happened that the jextract JVM was accidentally selected for compilation and I spend a fair amount of time finding out why many java.* packages could not be found.

**Comment by srikanth-sankaran** [comment]
Does this actually belong in jdt-core ? @jarthana ??

**Comment by iloveeclipse** [comment]
> Does this actually belong in jdt-core ? 

No, it is jdt-debug, and regression / side effect of https://github.com/eclipse-jdt/eclipse.jdt.debug/issues/230
